Source/Orts.Simulation/Common/Scripting/TrainControlSystem.cs

```diff
@@ -259,6 +292,12 @@ public abstract class TrainControlSystem : AbstractScriptClass
         /// </summary>
         public Action SetPantographsDown;
         /// <summary>
+        /// Cut power by pull all pantographs down.
+        /// PowerSupplyEvent: may be LowerPantograph or RaisePantograph
+        /// int: pantographID, from 1 to 4
+        /// </summary>
+        public Action<PowerSupplyEvent, int> SetPantograph;
```

Comment:
Currently, this is not right because you can use something inappropriate in the parameters (for exemple, StopEngine)

I would prefer something either more generic :
public Action<PowerSupplyEvent>
public Action<PowerSupplyEvent, int> SignalEvent;
Which is similar to the existing Action<Event> SignalEvent

Or more specific : 
public Action<int, bool> SetPantograph;
With the boolean indicating if we want the pantograph up or down.

Besides this, everything else looks fine to me.

Reply:
OK, I have created two specific hooks, one to raise the specified panto, and one to lower it.
I wasn't so happy with the boolean. I hope it is OK now.